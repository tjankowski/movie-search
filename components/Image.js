import React from "react";
import { union } from "utils";
import styles from "styles/Image.module.scss";

const NA = "N/A";

export default function Image({
  src = NA,
  className = styles.image,
  title = "",
}) {
  return (
    <div className={className}>
      {src !== NA ? (
        <>
          <img
            alt={`Blurred background for ${title}`}
            className={styles.image__blur}
            src={src}
          />
          <img alt={title} className={styles.image__poster} src={src} />
        </>
      ) : (
        <div className={styles.image__placeholder} />
      )}
    </div>
  );
}
