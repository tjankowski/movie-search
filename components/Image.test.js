import React from "react";
import renderer from "react-test-renderer";
import Image from "./Image";

describe("Image", () => {
  it("should render placeholder", () => {
    const component = renderer.create(<Image />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render image", () => {
    const component = renderer.create(<Image src="/empty.gif" />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should render image with class", () => {
    const component = renderer.create(
      <Image src="/empty.gif" className="class" />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
