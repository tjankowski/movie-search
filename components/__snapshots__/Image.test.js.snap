// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image should render image 1`] = `
<div
  className="image "
>
  <img
    className="image__blur"
    src="/empty.gif"
  />
  <img
    className="image__poster"
    src="/empty.gif"
  />
</div>
`;

exports[`Image should render image with class 1`] = `
<div
  className="image class"
>
  <img
    className="image__blur"
    src="/empty.gif"
  />
  <img
    className="image__poster"
    src="/empty.gif"
  />
</div>
`;

exports[`Image should render placeholder 1`] = `
<div
  className="image "
>
  <div
    className="image__placeholder"
  />
</div>
`;
